src/adapters/direction_api.js

```diff
@@ -74,9 +74,15 @@ export default class DirectionApi {
     try {
       response = await Ajax.get(directionsUrl, directionsParams, { timeout });
     } catch (e) {
-      return;
+      if (e >= 400 && e < 600) {
+        // Use the error codes 4xx and 5xx to display different error messages
+        response = { data: null, error: true, errorCode: e };
+      } else {
+        // Other errors are catched silently
+        return;
+      }
     }
-    if (directionConfig.api === 'qwant') {
+    if (directionConfig.api === 'qwant' && response.data) {
       response = response.data;
```

Comment:
Now that you established that the response shape is `{ data, error, errorCode }`, it would be better if the working case used it as well.
Otherwise you have now 3 different cases as result from this function:
- in case of a known error, an object `{ data: null, error: true, errorCode: … }`.
- in case of an unknown error, `undefined`
- in case of a working result, an object `{ routes: …, message: …, code: …}`
This makes managing all the cases more complex.

Reply:
ok, it will be just { data, error } then